improve sample.config.php parameter descriptions

christianlupus opened the pull request:
This is a replacement for nextcloud/documentation#9166. It must be solved here in the server repo.


* Resolves: # <!-- related github issue -->

## Summary
This just rearranges the sample config to obtain a better structure in terms of _sections_ in the documentation.

## TODO

- [ ] [One change](https://github.com/nextcloud/documentation/pull/9166/files#diff-f5b9c9ee7b52f546f39c5bb962cf7bc10bcacbad7972d5cc7a9373de9b9c7cf2L2539-R2542) was not yet incorporated

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included -> no changes made, so no tests written
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

christianlupus committed:
Reorder the sample config file in order to have a cleaner structure and better sectioning in the online manual

The change should only affect the conmments and reorder the config lines

Copy modifications from https://github.com/nextcloud/documentation/pull/9166 into sample config file

Signed-off-by: Christian Wolf <github@christianwolf.email>

kesselb commented:
Hi, can you use a self-explaining commit message to summarize the changes?

https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/

christianlupus commented:
I can offer to amend the commit with a new commit message. You mean to summarize the changes in the commit message, that it your point here, right?

kesselb commented:
> I can offer to amend the commit with a new commit message. 

Amend or rebase as you like.

> You mean to summarize the changes in the commit message, that it your point here, right?

Yes. "Copy modifications from ..." is not understandable without following the link. Please summarize what you did (e.g. "improve readability of config sample file ....") .

christianlupus committed:
Fix some issues that were not yet introduced in changes to sample config

This should not change anything but fix some documentation issues.
This change should not change anything in the implemenetation.

Signed-off-by: Christian Wolf <github@christianwolf.email>

szaimen commented on config/config.sample.php:
```diff
@@ -2291,9 +2288,13 @@
- *   \OCP\Accounts\IAccountManager::PROPERTY_PHONE => \OCP\Accounts\IAccountManager::SCOPE_PRIVATE
- * ]
+ * 
+ * ::
```
is this change correct?

christianlupus replied:
This should make that part a verbatim text. I can check what happens after doing the static parsing and code generation in sphinx. Let me have a look.

christianlupus replied:
I just built the documentation locally and looked at the output:
![image](https://user-images.githubusercontent.com/8202922/209777203-296ffd50-a009-4446-a034-ca0ce1463e15.png)

I do not see any issues here. What exactly were you referring to, @szaimen? The `::`? Or the indentation? Or something even different that I changed in fact without intention?

szaimen reviewed (approved):
LGTM, just one small question

christianlupus committed:
Fix comments in sample config to be correctly marked as verbatim

The comments were not marked as verbatim and thus the Sphinx compiler to generate
the online admin manual interpreted backslashes as special chars. They disappeared
from the output.
By adding ``, it is marked as verbatim and the backslashes are not removed.

Signed-off-by: Christian Wolf <github@christianwolf.email>
